File: location_app/lib/models/booking.dart
class Booking {
  final String id;
  final String propertyTitle;
  final String renterId;
  final String renterName;
  final String ownerId;
  String status; // pending, accepted, rejected
  final String message;
  final DateTime createdAt;
  
  Booking({
    required this.id,
    required this.propertyTitle,
    required this.renterId,
    required this.renterName,
    required this.ownerId,
    this.status = "pending",
    this.message = "",
    DateTime? createdAt,
  }) : this.createdAt = createdAt ?? DateTime.now();
}

File: backend/models/Booking.js
const mongoose = require('mongoose');

const bookingSchema = new mongoose.Schema({
  propertyId: { type: mongoose.Schema.Types.ObjectId, ref: 'Property', required: true },
  renterId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  renterName: { type: String, default: 'Locataire' },
  propertyTitle: { type: String },
  ownerId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  message: { type: String },
  startDate: { type: Date, required: true },
  endDate: { type: Date, required: true },
  status: { type: String, enum: ['pending', 'accepted', 'rejected'], default: 'pending' },
  images: [String],
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Booking', bookingSchema);
